docs/writing-tests/test-runner-with-vitest.mdx

```diff
@@ -0,0 +1,380 @@
+---
+title: 'Test runner with Vitest'
+sidebar:
+  order: 2
+  title: Test runner with Vitest
+---
+
+(⚠️ **Experimental**)
+
+Storybook's test runner with Vitest uses a Vitest plugin to transform your [stories](../writing-stories/index.mdx) into tests. You can then run those tests just like any other in Vitest, which will check that the story renders without errors and, if a [play function](../writing-stories/play-function.mdx) is defined, that it runs as expected and any [assertions made](../writing-tests/interaction-testing.mdx#assert-tests-with-vitests-apis) within it are validated.
+
+By using Vitest's browser mode, those tests are run in a real browser environment, which gives you more reliable results for UI components that commonly rely on browser APIs or features.
+
+## Setup
+
+Get started by installing and configuring the plugin in your project.
+
+### Automatic
+
+Run the following command to install and configure the addon, which contains the plugin to run your stories as tests using Vitest:
+
+{/* TODO: Snippetize */}
+```sh
+npx storybook@latest add @storybook/experimental-addon-vitest
```

Comment:
Didn't we had an agreement that `npx storybook add` is better for most command except upgrade and init? cc @shilman @valentinpalkovic have maybe more context here.